packages/pg-gateway/test/util.ts

```diff
@@ -128,3 +130,45 @@ export class DisposablePgClient extends Client {
     await this.end();
   }
 }
+
+export async function createPostgresClient(config: string | ClientConfig) {
+  const client = new DisposablePgClient(config);
+  await client.connect();
+  return client;
+}
+
+export class DisposableServer extends Server {
+  async [Symbol.asyncDispose]() {
+    await new Promise((resolve, reject) => {
+      this.close((err) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(undefined);
+        }
+      });
+    });
+  }
+}
```

Comment:
`net.Server` should already be async disposable - is it not?

Reply:
As it's a stage 3 proposal, I don't think any Node.js resources are disposable today. It's all compiled/transformed by TypeScript to me.

Reply:
They added experimental support in 18:

https://nodejs.org/api/net.html#serversymbolasyncdispose

But if we want to support/test pre-18 then we should keep this.